components/widget/widget_button.lua

```diff
@@ -0,0 +1,69 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Widget/Button
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Widget = Lua.import('Module:Widget')
+
+---@class ButtonWidgetParameters: WidgetParameters
+---@field title string
+---@field link string?
+---@field variant 'primary'|'secondary'|'ghost'|nil
+---@field size 'sn'|'md'|'lg'|nil
+
+---@class ButtonWidget: Widget
+---@operator call(ButtonWidgetParameters): ButtonWidget
+---@field title string
+---@field link string?
+---@field variant 'primary'|'secondary'|'ghost'
+---@field size 'sn'|'md'|'lg'
+
+local Button = Class.new(
+	Widget,
+	function(self, input)
+		self.title = self:assertExistsAndCopy(input.title)
+		self.link = input.link
+		self.variant = input.variant or 'primary' -- TODO: Validate
+		self.size = input.size or 'md' -- TODO: Validate
+	end
+)
+
+---@param children string[]
+---@return string
+function Button:make(children)
+	--- MW Parser does not allowed the <button> tag, so we use a div for now instead
+	local button = mw.html.create('div'):addClass('btn'):addClass('btn-new')
+	button:attr('title', self.title)
+	button:attr('aria-label', self.title)
+	button:attr('role', 'button')
+	button:attr('tabindex', '0')
```

Comment:
is there a reason to always set 0 here?
would we want/need other numbers here eventually?

Reply:
Needs to have tabindex=0 for dib buttons to be accessible by using tab 🤷

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex

Effectively, 0=accessible with tab, -1=not accessible, don't use any other number.

Buttons always have an interaction, so should always be accessible, and hence 0